File: Python/insertionsort.py
def insertionsort(list):
    l=len(list)
    for i in range(1,l):
        temp=list[i]
        j=i
        while j>0 and list[j-1]>temp:
            list[j]=list[j-1]
            j=j-1
        list[j]=temp
    return list

n=input("Enter a list : ")
i=insertionsort(n)
print i

File: C/arrays/insertionsort.c
#include<stdio.h>
int main()
{
	int a[20],i,j,n,temp;
	printf("Enter a number : ");
	scanf("%d",&n);
	for(i=0;i<n;i++)
	{
		printf("Enter a value : ");
		scanf("%d",&a[i]);

	}

	for(i=1;i<n;i++)
	{
		temp=a[i];
		j=i;
		while(j>0 && a[j-1]<temp)
		{
			a[j]=a[j-1];
			j=j-1;

		}
		a[j]=temp;

	}
	for(i=0;i<n;i++)
	{
		printf("%d\t",a[i]);
	}


}
